template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/zeppelin/ZeppelinS3CloudStorageRoleConfigProviderTest.java

```diff
@@ -0,0 +1,139 @@
+    }
+
+    @Test
+    public void testGetZeppelinStorageRoleConfigsWhenDefaultUser() {
+        TemplatePreparationObject preparationObject = getTemplatePreparationObject(true, false, true);
+        String inputJson = getBlueprintText("input/clouderamanager-ds.bp");
+        CmTemplateProcessor cmTemplateProcessor = new CmTemplateProcessor(inputJson);
+        Map<String, List<ApiClusterTemplateConfig>> roleConfigs = underTest.getRoleConfigs(cmTemplateProcessor, preparationObject);
+
+        List<ApiClusterTemplateConfig> zeppelinStorageConfigs = roleConfigs.get("zeppelin-ZEPPELIN_SERVER-BASE");
+        assertEquals(2, zeppelinStorageConfigs.size());
+
+        assertEquals("zeppelin-conf/zeppelin-site.xml_role_safety_valve", zeppelinStorageConfigs.get(0).getName());
+        String expected = "<property><name>zeppelin.notebook.s3.bucket</name><value>"
+                + getZeppelinS3Bucket().getValue()
+                + "</value></property>"
+                + "<property><name>zeppelin.notebook.s3.user</name><value>zeppelin</value></property>";
+        assertEquals(expected, zeppelinStorageConfigs.get(0).getValue());
+
+        assertEquals("org.apache.zeppelin.notebook.repo.S3NotebookRepo", zeppelinStorageConfigs.get(1).getValue());
+    }
+
+    @Test
+    public void testIsConfigurationNeededWhenS3FileSystem() {
+        TemplatePreparationObject preparationObject = getTemplatePreparationObject(true, true, true);
+        String inputJson = getBlueprintText("input/clouderamanager-ds.bp");
+        CmTemplateProcessor cmTemplateProcessor = new CmTemplateProcessor(inputJson);
+        boolean configurationNeeded = underTest.isConfigurationNeeded(cmTemplateProcessor, preparationObject);
+        assertTrue(configurationNeeded);
+    }
+
+    @Test
+    public void testIsConfigurationNeededWhenNotS3FileSystem() {
+        TemplatePreparationObject preparationObject = getTemplatePreparationObject(true, true, false);
+        String inputJson = getBlueprintText("input/clouderamanager-ds.bp");
+        CmTemplateProcessor cmTemplateProcessor = new CmTemplateProcessor(inputJson);
+        boolean configurationNeeded = underTest.isConfigurationNeeded(cmTemplateProcessor, preparationObject);
+        assertFalse(configurationNeeded);
+    }
+
+    private TemplatePreparationObject getTemplatePreparationObject(boolean includeLocations, boolean includeUser, boolean useS3FileSystem) {
+        HostgroupView master = new HostgroupView("master", 1, InstanceGroupType.GATEWAY, 1);
+        HostgroupView worker = new HostgroupView("worker", 2, InstanceGroupType.CORE, 2);
+
+        List<StorageLocationView> locations = new ArrayList();
+
+        if (includeLocations) {
+            locations.add(new StorageLocationView(getZeppelinS3Bucket()));
+        }
+
+        if (includeUser) {
+            locations.add(new StorageLocationView(getZeppelinS3User()));
```

Comment:
Seems like `includeBucket` would be a better name for `includeLocations`.

Reply:
updated name

Reply:
renamed